### DISCUSSION TITLE
Dashboard: Load Twig Template like in the Field Builder

### DISCUSSION THREAD

**Discussion by alexanderbuergin** [craftcms/cms#14184] [Ideas] [answered]
Currently, the Craft CMS dashboard does not offer a direct way to integrate customizable templates, similar to the Field Manager of Entry Types or Users. Users who want to personalize the dashboard currently have to resort to modules or plugins. However, this method can be cumbersome and overkill for simpler customizations.

Proposed solution
I propose implementing a functionality in Craft CMS that allows users to load custom templates directly into the dashboard. This feature should provide a simple interface for users to upload their own templates into s widget.

Advantages
1. flexibility: users can customize the dashboard according to their wishes and specific requirements.
2. simplicity: no need to develop your own modules or plugins for basic customizations.
3. user-friendliness: An intuitive interface allows even less technically experienced users to make changes.

Implementation
- A new section in the dashboard for managing your own templates.
- Support for common front-end technologies (HTML, CSS, JavaScript).
- Option to preview customizations before they go live.

The aim is to make the Craft CMS dashboard more customizable and user-friendly while reducing the need for complex solutions for simple customizations.

---

<img width="1738" alt="Bildschirmfoto 2024-01-23 um 23 41 56" src="https://github.com/craftcms/cms/assets/85960512/583c2e80-85ae-44ec-997d-7546435eb5dc">

<img width="1388" alt="Bildschirmfoto 2024-01-23 um 23 42 31" src="https://github.com/craftcms/cms/assets/85960512/a00b956a-e1a4-423a-8868-f749b8bae261">

**Comment by piotrpog** [comment]
Very good idea, +1

**Reply by alexanderbuergin** [reply]
+1

**Comment by brandonkelly** [comment]
I like the idea too. We could probably pull it off without any additional UI, just by designating a folder that custom widget templates should live in (probably `widgets/` at the root of your project by default).

The widgets could be Twig files that include YAML frontmatter for defining the widget name and icon:

```markdown
---
label: Welcome Widget
icon: 'hand-wave'
---

## Hello {{ currentUser.friendlyName }}

Welcome to your Craft CMS install…
```

**Reply by alexanderbuergin** [reply]
I think this solution is very good because of its simplicity.

**Reply by alexanderbuergin** [reply]
@brandonkelly when do you are planing to release that option?

**Reply by brandonkelly** [reply]
We should be able to add it in a 5.x release.

**Reply by alexanderbuergin** [reply]
@brandonkelly: Any Update on that one?

**Comment by brandonkelly** [comment, marked as answer]
We’ve added this for Craft 6 🎉  (#19319)
